client/dashboard/domains/domain-contact-verification/index.tsx

```diff
@@ -0,0 +1,235 @@
+import { PageHeader } from '../../components/page-header';
+import PageLayout from '../../components/page-layout';
+import { SectionHeader } from '../../components/section-header';
+import { findRegistrantWhois } from '../../utils/domain';
+
+const MAX_FILE_SIZE = 5 * 1024 * 1024; // 5MB
+
+export default function DomainContactVerification() {
+	const { domainName } = domainRoute.useParams();
+	const [ selectedFiles, setSelectedFiles ] = useState< FileList | null >( null );
+	const { data: domainWhois } = useSuspenseQuery( domainWhoisQuery( domainName ) );
+	const registrantWhoisData = findRegistrantWhois( domainWhois );
+	const [ submitted, setSubmitted ] = useState( false );
+	const [ error, setError ] = useState( false );
+	const { mutate: domainContactVerification, isPending: isSubmitting } = useMutation(
+		domainContactVerificationMutation( domainName )
+	);
+	const { createErrorNotice, createSuccessNotice } = useDispatch( noticesStore );
+	const router = useRouter();
+
+	if ( ! registrantWhoisData ) {
+		return null;
+	}
+
+	const handleFilesSelected = ( event: React.ChangeEvent< HTMLInputElement > ) => {
+		const files = event.currentTarget.files;
+		if ( ! files ) {
+			setSelectedFiles( null );
+			return;
+		}
+
+		if ( files.length > 3 ) {
+			createErrorNotice( __( 'You can only upload up to three documents.' ), { type: 'snackbar' } );
+			setSelectedFiles( null );
+			return;
+		}
+
+		const oversizedFiles = [ ...files ].filter( ( file: File ) => file.size > MAX_FILE_SIZE );
+		if ( oversizedFiles.length > 0 ) {
+			createErrorNotice( __( 'Files must be under 5MB each.' ), { type: 'snackbar' } );
+			setSelectedFiles( null );
+			return;
+		}
+
+		setSelectedFiles( files );
+	};
+
+	const renderSelectedFileList = () => {
+		if ( ! selectedFiles || selectedFiles.length === 0 ) {
+			return <Text variant="muted">{ __( 'No selected files yet' ) }</Text>;
+		}
+
+		const fileList = [ ...selectedFiles ].map( ( file: File ) => (
+			<li key={ file.name }>{ file.name }</li>
+		) );
+		return (
+			<>
+				<Text as="p">{ __( 'Selected files:' ) }</Text>
+				<ul>{ fileList }</ul>
+			</>
+		);
+	};
+
+	const submitFiles = () => {
+		if ( ! selectedFiles || selectedFiles.length === 0 ) {
+			createErrorNotice( __( 'Please select at least one file.' ), { type: 'snackbar' } );
+			return;
+		}
+
+		const formData: [ string, File, string ][] = [];
+		[ ...selectedFiles ].forEach( ( file: File ) => {
+			formData.push( [ 'files[]', file, file.name ] );
+		} );
+		domainContactVerification( formData, {
+			onSuccess: () => {
+				createSuccessNotice( __( 'Files submitted successfully.' ), { type: 'snackbar' } );
+				setSubmitted( true );
+			},
+			onError: () => {
+				createErrorNotice( __( 'Failed to submit files.' ), { type: 'snackbar' } );
+				setError( true );
+			},
+		} );
+	};
+
+	const renderSubmittedMessage = () => {
+		return (
+			<Notice status="success">
```

Comment:
I think we should be consistent between screens and use Notice from the `dashboard/components/notice` path.

Reply:
+1